src/reflect.rs

```diff
@@ -550,15 +542,47 @@ impl<'a> ReflectIntermediate<'a> {
                     _ => return Err(Error::UNSUPPORTED_TY),
                 };
                 let var = Variable::Descriptor(desc_bind, desc_ty);
-                if self.var_map.insert(op.alloc_id, var).is_some() {
-                    return Err(Error::ID_COLLISION);
-                }
-                // Leak out unknown types of uniform constants.
+                Some(var)
             },
             _ => {
                 // Leak out unknown storage classes.
+                None
             },
+        };
+
+        if let Some(var) = var {
+            // Determine descriptor access type.
+            let access_ty = match &var {
+                Variable::Descriptor(_, desc_ty) => {
+                    match desc_ty {
+                        DescriptorType::Image(..) | DescriptorType::StorageBuffer(..) => {
+                            let read_only = self.contains_deco(op.alloc_id, None, Decoration::NonWritable);
+                            let write_only = self.contains_deco(op.alloc_id, None, Decoration::NonReadable);
+                            match (read_only, write_only) {
+                                (true, true) => return Err(Error::ACCESS_CONFLICT),
```

Comment:
I think this might technically be allowed? AFAIK you are still allowed to query the image size and such even if the descriptor is marked non-readable and non-writable.

Reply:
It is an obscure case though (and not one I have any use for personally) so it may be worth ignoring anyway

Reply:
Well, I thought about it before. But my reasoning was that, in the SPIR-V specification, `NonWritable` and `NonReadable` are discussed together with `Coherent` and `Volatile`, which are mapped directly to the corresponding GLSL keywords to decorate storage resources. Even though it's not prohibited by the SPIR-V spec, the GLSL spec doesn't provide an interface to make a storage resource both `NonWritable` and `NonReadable`. Added to it that I was hesitating adding a new variant to `AccessType` to represent such unusual resource usage, I decided to keep it invalid until someone actually needs it.

I would keep this in mind though. Thanks for the comments. 😀